File: 732-my-calendar-iii/732-my-calendar-iii.py
class MyCalendarThree:

    def __init__(self):
        self.li = []

    def book(self, start: int, end: int) -> int:
        bisect.insort(self.li, (start, 1))
        bisect.insort(self.li, (end, -1))
        
        res, cnt = 0, 0
        for pos, n in self.li:
            cnt += n
            res = max(res, cnt)
        
        return res

# Your MyCalendarThree object will be instantiated and called as such:
# obj = MyCalendarThree()
# param_1 = obj.book(start,end)

File: 732-my-calendar-iii/732-my-calendar-iii.cpp
class MyCalendarThree {
public:
    MyCalendarThree() {
        
    }
    
    map<int, int> bmap;
    int book(int start, int end) {
        bmap[start]++;
        bmap[end]--;
        
        int res = 0, sum = 0;
        for(auto &p : bmap){
            sum += p.second;
            res = max(res, sum);
        }
        
        return res;
    }
};

/**
 * Your MyCalendarThree object will be instantiated and called as such:
 * MyCalendarThree* obj = new MyCalendarThree();
 * int param_1 = obj->book(start,end);
 */